qemu/cpus.c

```diff
@@ -191,15 +169,10 @@ static int qemu_tcg_init_vcpu(CPUState *cpu)
         uc->tcg_halt_cond = cpu->halt_cond;
         snprintf(thread_name, VCPU_THREAD_NAME_SIZE, "CPU %d/TCG",
                 cpu->cpu_index);
-        if (qemu_thread_create(uc, cpu->thread, thread_name, qemu_tcg_cpu_thread_fn,
-                cpu, QEMU_THREAD_JOINABLE))
-            return -1;
+        qemu_thread_get_self(uc, cpu->thread);
```

Comment:
I looked at that loop, which in turn calls tcg_cpu_exec -> cpu_exec which contains this infinite loop:

https://github.com/lunixbochs/unicorn/blob/master/qemu/cpu-exec.c#L100

which only seems to break on a stop_request or an exception/interrupt. If that is the case, any other CPUs would locked out until the current CPU hits a stop condition.

I agree that more thought needs to be put into a solution that supports multiple CPUs